src/test/java/io/reactivex/internal/operators/flowable/FlowableFromAsyncTest.java

```diff
@@ -0,0 +1,729 @@
+package io.reactivex.internal.operators.flowable;
+
+import org.junit.*;
+import org.reactivestreams.*;
+
+import io.reactivex.*;
+import io.reactivex.exceptions.*;
+import io.reactivex.functions.Consumer;
+import io.reactivex.processors.PublishProcessor;
+import io.reactivex.subscribers.*;
+
+public class FlowableFromAsyncTest {
+
+    PublishAsyncEmitter source;
+
+    PublishAsyncEmitterNoCancel sourceNoCancel;
+
+    TestSubscriber<Integer> ts;
+    
+    @Before
+    public void before() {
+        source = new PublishAsyncEmitter();
```

Comment:
nit: you can inline it and make variables final, JUnit creates new instance of test class for each test